v2/client/client.go

```diff
@@ -152,6 +152,10 @@ func (c *ceClient) Request(ctx context.Context, e event.Event) (*event.Event, pr
 		return nil, err
 	}
 
+	// Event has been defaulted and validated, record we are going to perform request.
+	ctx, cb := c.observabilityService.RecordRequestEvent(ctx, e)
```

Comment:
same as above

Reply:
I moved this because if we do not do the defaulting first, the recorder does not see the true event on the wire and you do not get to record the `id`